lib/elements/element.py

```diff
@@ -150,16 +150,26 @@ def set_param(self, name, value):
     @property
     @cache
     def style(self):
-        declarations = tinycss2.parse_declaration_list(self.node.get("style", ""))
-        style = {declaration.lower_name: declaration.value[0].serialize() for declaration in declarations}
+        return self.parse_style()
 
+    def parse_style(self, node=None):
+        if node is None:
+            node = self.node
+        declarations = tinycss2.parse_declaration_list(node.get("style", ""))
+        style = {declaration.lower_name: declaration.value[0].serialize() for declaration in declarations}
         return style
 
     def get_style(self, style_name, default=None):
```

Comment:
And we should definitely add `@cache` for this one too, since it's now a fairly laborious process to figure out an element's style.